### PR TITLE
Drop unnecessary swizzling

### PR DESCRIPTION
**Pull request by LittleRockInGitHub**
Hi, I think the swizzling is unnecessary. The result is able to be generated on `NSTextCheckResult` lazily.
Since no swizzling, there is no need for objc. And it's better store such info on `NSRegularExpression`, since it's all about pattern.
I tried refactoring with less locs and all tests passed.

### CODE REVIEW

File: Sources/_Core.swift [hunk 1]
```diff
@@ -43,79 +42,16 @@ fileprivate let NamedCaptureGroupsPattern = try! NSRegularExpression(
...
-  public let _index: Int
-
-  @nonobjc fileprivate init( _ tuple: _GroupNamesSearchResult ) {
-    _outerOrdinaryCaptureGroup = tuple._outerOrdinaryCaptureGroup
-    _innerRefinedNamedCaptureGroup = tuple._innerRefinedNamedCaptureGroup
-    _index = tuple._index
-    }
-  }
 
 public extension NSRegularExpression /* _NamedCaptureGroupsSupport */ {
```
Comment (torinkwok) [inline Sources/_Core.swift]: Without having to do ObjC *mix and match*, there's no longer need to expose this extension broadly. Just dropping the `public` is okay.

File: Sources/NSTextCheckResult+NamedCapture.swift [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  public func rangeWith(_ name: String?) -> NSRange {
+    
+    guard let name = name else { return self.rangeAt(0) }
+    
+    return regularExpression?._namedCaptures[name].map { rangeAt($0) } ?? NSRange(location: NSNotFound, length: 0)
+  }
+  
+}
+
+extension NSRegularExpression {
```
Comment (torinkwok) [inline Sources/NSTextCheckResult+NamedCapture.swift]: This extension should be marked as `fileprivate`.

File: Sources/NSTextCheckResult+NamedCapture.swift [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  
+  private struct AssociatedKeys {
+    static var namedCaptures: String = "nc.namedCaptures"
+  }
+
+  var _namedCaptures: [String : Int] {
```
Comment (torinkwok) [inline Sources/NSTextCheckResult+NamedCapture.swift]: `fileprivate`, too.

File: Sources/NSTextCheckResult+NamedCapture.swift [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  /// optionally have more (for example, to represent regular
+  /// expression capture groups).
+  ///
+  /// @pram groupName The name of capture group that appears in the regex
+  ///       pattern. Passing the value `nil` if the overall range is expected.
+  ///
+  /// @return The range of the result.
+  ///         Passing the method the value `nil` always returns
+  ///         the value of the the `range` property. Additional ranges,
+  ///         if any, can be retrieved through their capture group names.
```
Comment (torinkwok) [inline Sources/NSTextCheckResult+NamedCapture.swift]: Our sole public API is Swifty 🐣 now, so I suppose the documentation markup should also be Swift-style.

**Review by torinkwok — CHANGES_REQUESTED** [review]
Hi Rock.

You've done a good job! I simply ignored that `NSTextCheckingResult` has a readonly property that references back to its regex instance, and, as a result, introduced a bunch of ugly hacks. That's so stupid indeed.

Anyway, I've marked several places where need some changes. I'll merge your p-r as soon as you have fixed them.

File: Sources/NSTextCheckResult+NamedCapture.swift [hunk 5]
```diff
@@ -0,0 +1,54 @@
+//
+//  NSTextCheckResult+NamedCapture.swift
```
Comment (torinkwok) [inline Sources/NSTextCheckResult+NamedCapture.swift]: I suppose the file name should be renamed as _NSTextCheckingResult+NamedCaptureGroup_.swift. Mouthful, but the consistency would be guaranteed.

**Review by torinkwok — APPROVED** [review]

File: Sources/NSTextCheckResult+NamedCapture.swift [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+//  NSRegExNamedCaptureGroup
+//
+//  Created by Rock Yang on 2017/6/24.
```
Comment (torinkwok) [inline Sources/NSTextCheckResult+NamedCapture.swift]: Yang -> Young? 😆

File: Sources/NSTextCheckResult+NamedCapture.swift (hunk 2 above)
Comment (LittleRockInGitHub) [reply Sources/NSTextCheckResult+NamedCapture.swift]: Typically, we declare access control on individual members rather than extension itself.

**Comment by LittleRockInGitHub** [comment]
Fixed

**Pull request merged**